lib/http2/hpack.c

```diff
@@ -651,19 +663,22 @@ int h2o_hpack_parse_response(h2o_mem_pool_t *pool, h2o_hpack_decode_header_cb de
             PARSE_DIGIT(1, 0);
 #undef PARSE_DIGIT
         } else {
-            if (status != NULL && *status == 0)
+            if (status != NULL && *status == 0) {
+                *err_desc = ":status not first header in response";
                 return H2O_HTTP2_ERROR_PROTOCOL;
+            }
             if (h2o_iovec_is_token(name)) {
                 h2o_token_t *token = H2O_STRUCT_FROM_MEMBER(h2o_token_t, buf, name);
                 /* reject headers as defined in draft-16 8.1.2.2 */
                 if (token->flags.is_hpack_special) {
-                    if (token == H2O_TOKEN_CONTENT_LENGTH || token == H2O_TOKEN_CACHE_DIGEST) {
+                    if (token == H2O_TOKEN_CONTENT_LENGTH || token == H2O_TOKEN_CACHE_DIGEST || token == H2O_TOKEN_HOST) {
                         /* pass them through when found in response headers (TODO reconsider?) */
                     } else if (token == H2O_TOKEN_DATAGRAM_FLOW_ID) {
                         if (datagram_flow_id != NULL)
                             *datagram_flow_id = value;
                         goto Next;
                     } else {
+                        *err_desc = "unexpected connection-specific header";
```

Comment:
I believe this is the line that would have helped you from figuring out the cause?

Reply:
Yes, that's it.